generator/lib/control.ml

```diff
@@ -46,6 +46,35 @@ let api =
           ];
         ]
       )) in
+  let configuration = {
+    Arg.name = "configuration";
+    ty = Type.(Dict(String, Basic String));
+    description = String.concat " " [
+                    "Plugin-specific configuration which describes where and";
+                    "how to locate the storage repository. This may include";
+                    "the physical block device name, a remote NFS server and";
+                    "path or an RBD storage pool.";
+                  ];
+  } in
+  let probe_result_decl =
+    Type.(Struct (
+        ( configuration.name, configuration.ty, configuration.description ),
+        [ "complete", Basic Boolean, String.concat " " [
+            "True if this configuration is complete and can be used to";
+            "call SR.create. False if it requires further iterative calls";
+            "to SR.probe, to potentially narrow down on a configuration";
+            "that can be used.";
+          ];
+          "sr", Option (Name "sr_stat"), String.concat " " [
+            "Existing SR found for this configuration";
```

Comment:
I assume this `sr` means there is an existing _attached_ SR, because the SR.create only returns a `configuration`, it's only SR.attach that returns an `sr`?

Reply:
It is an `attachable` SR, i.e. an SR that exists with the given config and you can attach it if you want to. It is not necessarily attached currently.

Reply:
Hmm, interesting, because as I said on https://github.com/xapi-project/xcp-idl/pull/208, if we consider SMAPIv3 itself, the interface allows us to use these srs returned by probe to do operations like SR.ls. :thinking:

Reply:
well, yes and also no. Theoretically you could use the URI to issue an SR.ls, but after the probe completes the mount point will disappear and your SR.ls will vanish. It might be sensible for SR.probe to blank out the uri field of any SRs it returns

Reply:
Yes, and that's the problem: if you just look at the types, we get back an `sr` even for SRs that are not attached, and that might make an API user think that they can use it with `SR.ls` for example, when in fact they cannot, because it isn't attached.

Reply:
but, the expectation is that this will give you the same result as an SR.stat, in terms of the SR name, description, size, utilisation etc. It's only the URi that iffy and we could just decree that we drop the URI from an returned srs